File: Schedule/Schedule.Core/Models/LessonTemplate.cs
namespace Schedule.Core.Models;

public class LessonTemplate
{
    public int LessonTemplateId { get; set; }

    public int Number { get; set; }

    public int? Subgroup { get; set; }

    public int? TimeId { get; set; }

    public int TemplateId { get; set; }

    public int? DisciplineId { get; set; }

    public virtual Discipline? Discipline { get; set; }

    public virtual ICollection<LessonTemplateTeacherClassroom> LessonTemplateTeacherClassrooms { get; set; } =
        new List<LessonTemplateTeacherClassroom>();

    public virtual Template Template { get; set; } = null!;

    public virtual Time? Time { get; set; }
}

File: api.schedule/schedule-main-service/internal/model/lesson_template.go
package model

import "time"

type LessonTemplate struct {
	ID           int64
	Number       int64
	Start        time.Time
	End          time.Time
	Duration     time.Time
	TemplateID   int64
	DisciplineID int64
	TeacherIDs   []int64
	ClassroomIDs []int64
}
